src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -501,22 +534,41 @@ function relabel_visited_cell!(bsn_nfo::BasinsInfo, old_label, new_label)
 end
 
 function basin_cell_index(y_grid_state, bsn_nfo::BasinsInfo{B}) where {B}
-    iswithingrid = true
-    @inbounds for i in eachindex(bsn_nfo.grid_minima)
-        if !(bsn_nfo.grid_minima[i] ≤ y_grid_state[i] ≤ bsn_nfo.grid_maxima[i])
-            iswithingrid = false
-            break
+      
+    if bsn_nfo.grid_nfo isa RegularGrid
+
+        iswithingrid = true
+
+        @inbounds for i in eachindex(bsn_nfo.grid_nfo.grid_minima)
+
+            if !(bsn_nfo.grid_nfo.grid_minima[i] ≤ y_grid_state[i] ≤ bsn_nfo.grid_nfo.grid_maxima[i])
+                iswithingrid = false
+                break
+            end
         end
+        if iswithingrid
+            # Snap point to grid
+            ind = @. round(Int, (y_grid_state - bsn_nfo.grid_nfo.grid_minima)/bsn_nfo.grid_nfo.grid_steps) + 1
+            return CartesianIndex{B}(ind...)
+        else
+            return CartesianIndex{B}(-1)
+        end
+    else 
+        for (axis, coord) in zip(bsn_nfo.grid_nfo.grid, y_grid_state)
+            if coord < minimum(axis) || coord > maximum(axis)
+                return CartesianIndex{B}(-1)
+            end
+        end
+
+        cell_indices = map(x -> searchsortedlast(x[1], x[2]), zip(bsn_nfo.grid_nfo.grid, y_grid_state))
```

Comment:
don't write `x` and `x[1], x[2]`. Write `x, y` and use `x, y` in `searchsortedlast`.